src/python/pants/help/help_printer.py

```diff
@@ -148,7 +154,22 @@ def _format_help(self, scope_info):
       show_advanced=self._help_request.advanced,
       color=sys.stdout.isatty(),
     )
+
+    # TODO(#8884): Figure out how to traverse the rule graph to find all `subsystem_rules` used
+    #  transitively by the Optionable in question.
```

Comment:
I have no idea where to begin with this part.

Reply:
It would start with calling https://github.com/pantsbuild/pants/blob/1af739dbd4a04fc045085e93b41da9b101bc0d19/src/rust/engine/rule_graph/src/lib.rs#L826-L830 with arguments with the same shape as the `run_goal_rule` method (ie, the `OptionsBootstrapper`, `Specs`, etc), and then transitively walking the `RuleEdges` to collect all of the involved types. Depending on how you wanted to display them, the interface across the rust boundary could be "list of transitive dependency types" or something.